### ISSUE TITLE
Potential namespace conflict with ts.el

### ISSUE THREAD

**Issue by alphapapa** [ubolonton/emacs-tree-sitter#35]
Hi @ubolonton,

Following the recent discussions on emacs-devel, I saw you mention a `ts-parse` function.  I can't find that function's definition in the `master` branch, but I do see some calls to it.  So I thought I should mention that there is a potential namespace conflict with my library https://github.com/alphapapa/ts.el, which implements a `ts-parse` function.

Thanks for your work on this package.  I'm looking forward to the improvements it will bring to Emacs.

**Comment by ubolonton** [comment]
Thanks for the kind words!

`ts-parse` is defined by the dynamic module [here](https://github.com/ubolonton/emacs-tree-sitter/blob/81bcbcd/src/parser.rs#L53).

I think I'll rename it to something else, maybe `ts-parse-chunks`. It may be better if my package switches to another prefix, but I can't think of a satisfactory one currently.

**Comment by alphapapa** [comment]
Naming things is hard.  :)  Thanks.

**Comment by sogaiu** [comment]
On a side note, does the `tree-sitter-dyn.so` that's mentioned [here](https://github.com/ubolonton/emacs-tree-sitter#setup-for-development) need to be updated to account for this change?

**Comment by ubolonton** [comment]
> On a side note, does the tree-sitter-dyn.so that's mentioned here need to be updated to account for this change?

Yes, but I haven't uploaded the new version yet, since I intend to add a mechanism for the binaries to be versioned and downloaded separately first, so that the package as a whole can be published on MELPA.

Do you need this change soon?

**Comment by sogaiu** [comment]
Thanks for the status update.

Initially I didn't know how to build it and put the result in place, but that turned out to be doable, so I'm all set :)

Wanted to try the query builder and that now seems to work:

![tree-sitter-query-builder-with-clojure](https://user-images.githubusercontent.com/33044872/78608545-09cbe000-7859-11ea-8a89-8d774af24f21.png)

**Comment by shackra** [comment]
> Wanted to try the query builder

I made most of that feature for several months, any feed back? :D

**Comment by sogaiu** [comment]
Below are some things I noticed about the version I tried at the time of the screenshot (I went back and tried it again just now -- and I haven't updated since that time):

* I thought it was helpful in making it clear what matched.
* I've used it on one of Clojure's larger source files ([core.clj](https://github.com/clojure/clojure/blob/master/src/clj/clojure/core.clj) 263653 bytes) and it seemed to perform without any difficulty so that's definitely nice.
* The echo area was a bit noisy while the query is being constructed and not complete (Error: (rust-panic "called ...) etc.), but I can understand at this stage that it would be helpful to have that information (at least if you are the one developing the query builder).

Overall I think it is nice for testing out queries as well as for helping to interactively develop one's mental model of how tree-sitter queries work.

Thanks for making it!
